platform/darwin/src/MGLShapeSource.mm

```diff
@@ -84,6 +86,52 @@
         geoJSONOptions.cluster = value.boolValue;
     }
 
+    if (NSDictionary *value = options[MGLShapeSourceOptionClusterProperties]) {
+        if (![value isKindOfClass:[NSDictionary<NSString *, NSArray *> class]]) {
+            [NSException raise:NSInvalidArgumentException
+                        format:@"MGLShapeSourceOptionClusterProperties must be an NSDictionary with an NSString as a key and an array containing two NSExpression objects as a value."];
+        }
+
+        NSEnumerator *stringEnumerator = [value keyEnumerator];
+        NSString *key;
+
+        while (key = [stringEnumerator nextObject]) {
+            // check that array has only 2 values
+            NSArray *expArray = value[key];
```

Comment:
Is `expArray` guaranteed to be an array?
Same for `expArray[0]` and `expArray[1]`